### PR TITLE
feat: support multiple names for key_mapping with kubernetes secrets

### PR DESCRIPTION
**Pull request by leafty**
Part of #509.

Add support for multiple names in `key_mapping` when creating a kubernetes secret. This allows the same user secret to be supplied more than once in sessions.

### CODE REVIEW

File: components/renku_data_services/secrets/core.py [hunk 1]
```diff
@@ -44,10 +44,19 @@ async def create_k8s_secret(
     if len(missing_secret_ids) > 0:
         raise errors.MissingResourceError(message=f"Couldn't find secrets with ids {', '.join(missing_secret_ids)}")
 
-    if key_mapping:
-        if set(key_mapping) != requested_secret_ids:
+    def ensure_list(value: str | list[str]) -> list[str]:
+        return [value] if isinstance(value, str) else value
+
+    key_mapping_list = {key: ensure_list(key_mapping[key]) for key in key_mapping} if key_mapping else None
```
Comment (Panaetius) [inline components/renku_data_services/secrets/core.py:50]: nitpick: it threw me off a bit at first glance that this is called `key_mapping_list` when it's a dict

File: components/renku_data_services/secrets/core.py [hunk 2]
```diff
@@ -44,10 +44,19 @@ async def create_k8s_secret(
...
+
+    if key_mapping_list:
+        if set(key_mapping_list) != requested_secret_ids:
```
Comment (Panaetius) [inline components/renku_data_services/secrets/core.py:53]:
```suggestion
        if key_mapping_list.keys() != requested_secret_ids:
```

File: components/renku_data_services/secrets/core.py [hunk 3]
```diff
@@ -44,10 +44,19 @@ async def create_k8s_secret(
...
             raise errors.ValidationError(message="Key mapping must include all requested secret IDs")
-        if len(key_mapping) != len(set(key_mapping.values())):
+
+        all_keys: list[str] = []
+        for value in key_mapping_list.values():
+            all_keys.extend(value)
```
Comment (Panaetius) [inline components/renku_data_services/secrets/core.py:58]: suggestion:
```suggestion
        all_keys = [key for value in key_mapping_list.values() for key in value]
```

**Review by Panaetius — CHANGES_REQUESTED** [review]

File: components/renku_data_services/secrets/core.py (hunk 1 above)
Comment (leafty) [reply components/renku_data_services/secrets/core.py:50]: Do you have a suggestion? The variable is `key_mapping` except we ensure all values are `list[str]`.

Comment (Panaetius) [reply components/renku_data_services/secrets/core.py:50]: not really, maybe something like aggregated_key_mappings

**Review by Panaetius — APPROVED** [review]

**Pull request merged**